instrumentation/rocketmq-client-4.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rocketmq/RocketMqClientApiImplInstrumentation.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.rocketmq;
+
+import static io.opentelemetry.instrumentation.rocketmq.RocketMqProducerTracer.tracer;
+import static io.opentelemetry.instrumentation.rocketmq.TextMapInjectAdapter.SETTER;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.rocketmq.SendCallbackWrapper;
+import io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.rocketmq.client.producer.SendCallback;
+import org.apache.rocketmq.common.message.Message;
+import org.apache.rocketmq.common.protocol.header.SendMessageRequestHeader;
+
+public class RocketMqClientApiImplInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("org.apache.rocketmq.client.impl.MQClientAPIImpl");
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        isMethod().and(named("sendMessage")).and(takesArguments(12)),
+        RocketMqClientApiImplInstrumentation.class.getName() + "$SendMessageAdvice");
+  }
+
+  public static class SendMessageAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnter(
+        @Advice.Argument(value = 0, readOnly = false) String addr,
+        @Advice.Argument(value = 2, readOnly = false) Message msg,
+        @Advice.Argument(value = 3, readOnly = false) SendMessageRequestHeader requestHeader,
+        @Advice.Argument(value = 6, readOnly = false) SendCallback sendCallback,
+        @Advice.Local("otelSpan") Span span,
+        @Advice.Local("otelScope") Scope scope) {
+
+      Context parent = Java8BytecodeBridge.currentContext();
+      span = tracer().startProducerSpan(addr, msg);
+      Context newContext = parent.with(span);
+      try {
+        Java8BytecodeBridge.getGlobalPropagators()
+            .getTextMapPropagator()
+            .inject(newContext, requestHeader, SETTER);
```

Comment:
Hmm, wouldn't it be worth it to check `RocketMqClientConfig.isPropagationEnabled()` here as well?